lib/cors-anywhere.js

```diff
@@ -11,6 +11,34 @@ var getProxyForUrl = require('proxy-from-env').getProxyForUrl;
 
 var help_text = {};
 
+/**
+ * Converts a hostname to an environment variable name.
+ * @param {string} hostname - The hostname to convert.
+ * @returns {string} - The converted environment variable name.
+ */
+function convertHostnameToEnvVarName(hostname) {
+  // Because environment variables do not allow dots in their names, 
+  // we convert hostnames to environment variable names.
+  // E.g for 'www.example.com' we expect the env key name to be: 'SECRET_KEY_WWW_DOT_EXAMPLE_DOT_COM'
+  return 'SECRET_KEY_' + hostname.toUpperCase().replace(/\./g, '_DOT_');
+}
+
+/**
+ * Injects a secret key into the request headers if exists.
+ * @param {object} reqHeaders - The request headers to inject the secret key into.
+ * @param {string} locationHostname - The hostname of the location being requested.
+ */
+function injectSecretKey(reqHeaders, locationHostname) {
+  // If a secret key is set as an environment variable for the requested host, 
+  // we get it so we can inject it into the request headers. 
+  const envVarName = convertHostnameToEnvVarName(locationHostname);
+  const secretKey = process.env[envVarName];
+
+  if (secretKey) {
+    reqHeaders['Authorization'] = secretKey;
```

Comment:
This could alternately be `reqHeaders.Authorization = secretKey;`

Reply:
Yes but when it comes time to set an arbitrary key it would need changing back again to the same setting method wouldn't it?

Reply:
Yes, it would need bracket notation for a string literal. But if it was always going to be a key named Authorization then dot notation would be functionally equivalent I think.